### PR TITLE
Update ResNet50 example to work with TensorFlow 2.x

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It updates ResNet50 example to work with TensorFlow 2.x

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     updates ResNet50 example to work with TensorFlow 2.x
 - Affected modules and functionalities:
     ResNet50 TensorFlow example
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI
 - Documentation (including examples):
     NA

**JIRA TASK**: *[DALI-1779]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: docs/examples/use_cases/tensorflow/resnet-n/nvutils/image_processing.py [hunk 1]
```diff
@@ -15,170 +14,316 @@
...
-            output_layout="HWC")
-        self.mirror = dali.ops.CoinFlip()
-        self.iter = 0
-
-    def define_graph(self):
-        # Read images and labels
-        inputs = self.input(name="Reader")
-        images = inputs["image/encoded"]
+            std=[58.4, 57.12, 57.3],
+            output_layout=types.NHWC,
```
Comment (jantonguirao) [inline docs/examples/use_cases/tensorflow/resnet-n/nvutils/image_processing.py:208]:
```suggestion
            output_layout="HWC",
```
Using layouts in `types.*` is deprecated

File: qa/TL3_RN50_convergence/test_tensorflow.sh [hunk 2]
```diff
@@ -38,16 +39,15 @@ if [[ $RET -ne 0 ]]; then
...
 
-MIN_TOP1=75.0
-MIN_TOP5=92.0
-MIN_PERF=9500
 
-TOP1=$(grep "^Top-1" $LOG | awk '{print $3}')
-TOP5=$(grep "^Top-5" $LOG | awk '{print $3}')
+MIN_TOP1=0.75
+MIN_TOP5=0.92
+MIN_PERF=8000
```
Comment (jantonguirao) [inline qa/TL3_RN50_convergence/test_tensorflow.sh:45]: why did we lower the threshold?

Comment (JanuszL) [reply qa/TL3_RN50_convergence/test_tensorflow.sh:45]: 1) TF2.0 seems to be slower
2) It is RN50 v1.5 which is a bit slower but more accurate AFAIK.
3) It consumes more memory and we need to run smaller batch which lowers the perf as well.

File: docs/examples/use_cases/tensorflow/resnet-n/README.rst [hunk 3]
```diff
@@ -4,34 +4,90 @@ ResNet-N with TensorFlow and DALI
...
-Here we have assumed that imagenet is stored in tfrecord format in the
-directory '/data/imagenet'. After training completes, evaluation is
-performed using the validation dataset.
-
-Some common training parameters can tweaked from the command line.
-Others must be configured within the network scripts themselves.
-
-Original scripts modified from ``nvidia-examples`` scripts in `NGC
-TensorFlow Container`_
+It implements the ResNet50 v1.5 CNN model and demonstrate efficient
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/use_cases/tensorflow/resnet-n/README.rst]: Fixed

File: docs/examples/use_cases/tensorflow/resnet-n/nvutils/image_processing.py (hunk 1 above)
Comment (JanuszL) [reply docs/examples/use_cases/tensorflow/resnet-n/nvutils/image_processing.py]: Copy-paste. Fixed

**Review by jantonguirao — APPROVED** [review]

**Review by banasraf — APPROVED** [review]

**Pull request merged**